src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -189,7 +189,17 @@ private SettingConfiguration convert(org.opensrp.domain.postgres.Settings settin
 		if (setting == null || setting.getJson() == null || !(setting.getJson() instanceof SettingConfiguration)) {
 			return null;
 		}
-		return (SettingConfiguration) setting.getJson();
+		SettingConfiguration settingConfiguration = (SettingConfiguration) setting.getJson();
+		
+		SettingsMetadataExample metadataExample = new SettingsMetadataExample();
+		metadataExample.createCriteria().andSettingsIdEqualTo(setting.getId());
+		
+		SettingsMetadata settingMetadata = settingMetadataMapper.selectByExample(metadataExample).get(0);
+		settingConfiguration.setId(settingMetadata.getSettingsId().toString());
+		settingConfiguration.setDocumentId(settingMetadata.getDocumentId());
```

Comment:
These properties should be stored in the JSON document so that this query can be avoided. The metadata should only used if one needs to for faster query using relational indexes. But the JSON document should contain all the information when being saved